apps/server/server.cpp

```diff
@@ -511,6 +505,155 @@ void invoke_sdk_api(payload::ClientRequest buff_recv) {
         break;
     }
 
+    case EEPROM_OPEN: {
+#ifdef DEBUG
+        cout << "EepromOpen function\n";
+#endif
+        // Check if client has sent exactly one name to identify the EEPROM
+        if (buff_recv.device_data().eeproms_size() != 1) {
+            buff_send.set_message(
+                "Failed to idenify EEPROM. Exactly one name for the eeprom is "
+                "accepted. Number of names provided: " +
+                std::to_string(buff_recv.device_data().eeproms_size()));
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
+            break;
+        }
+        std::string eName = buff_recv.device_data().eeproms(0).driver_name();
+
+        // If EEPROM hasn't been instantiated yet, we do it here
+        auto eeprom = findEepromByName(eName);
+        if (!eeprom) {
+            eeprom = std::shared_ptr<aditof::EepromInterface>(
+                aditof::EepromFactory::buildEeprom(
+                    aditof::ConnectionType::LOCAL));
+            if (!eeprom) {
+                buff_send.set_message("Failed to instantiate an EEPROM object");
+                buff_send.set_status(static_cast<::payload::Status>(
+                    aditof::Status::GENERIC_ERROR));
+                break;
+            }
+        }
+
+        aditof::Status status = eeprom->open(
+            nullptr, eName, buff_recv.device_data().eeproms(0).driver_path());
+
+        if (status == aditof::Status::OK) {
+            eeproms.push_back(eeprom);
+        }
+
+        buff_send.set_status(static_cast<::payload::Status>(status));
+        break;
+    }
+
+    case EEPROM_READ: {
+#ifdef DEBUG
+        cout << "EepromRead function\n";
+#endif
+        // Check if client has sent exactly one name to identify the EEPROM
+        if (buff_recv.device_data().eeproms_size() != 1) {
+            buff_send.set_message(
+                "Failed to idenify EEPROM. Exactly one name for the eeprom is "
+                "accepted. Number of names provided: " +
+                std::to_string(buff_recv.device_data().eeproms_size()));
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
+            break;
+        }
+        std::string eName = buff_recv.device_data().eeproms(0).driver_name();
+
+        // We lookup the EEPROM instance by name
+        auto eeprom = findEepromByName(eName);
+        if (!eeprom) {
+            buff_send.set_message("Failed to idenify EEPROM with name: " +
+                                  eName);
+            buff_send.set_status(static_cast<::payload::Status>(
+                aditof::Status::INVALID_ARGUMENT));
```

Comment:
missing a `break;`? if there is no eeprom this line `577: aditof::Status status = eeprom->read(address, buffer, length);` will dereference a nullptr?

Reply:
fixed